### PR TITLE
add revelation-0.5.3

### PR DESCRIPTION
**Pull request by misanthropos**
add updated ebuild for revelation-0.5.3 removing https://bugs.gentoo.org/708142

### CODE REVIEW

File: x11-misc/revelation/metadata.xml
Comment (juippis) [inline x11-misc/revelation/metadata.xml:4]: We **can't** add new packages without dedicated maintainers. 
https://projects.gentoo.org/qa/policy-guide/maintainer.html#pg0602 (New packages without maintainers)
Besides this package was removed because it didn't have a maintainer.

You can add yourself + proxy-maint project as described in here,
https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide#Proxied_maintainer_in_metadata.xml

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:5]: Please put an empty line between these files.

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:4]: Do you need to use *gnome2.eclass*? I know it sets some defaults for you, but you should be fine using simple *autotools.eclass* here.

You can use the latest EAPI-7 that we prefer if you get rid of outdated *gnome2.eclass*

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:5]: python-3.8 is the new default for us, please update.

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:10]: Please also add the Github page as 2ndary HOMEPAGE here.

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:13]: They seem to use `GPL-2+` instead.

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:15]: Reset the KEYWORDS to `~`.

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:17]: Empty variable can be dropped with EAPI-7.

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:20]: Why is this restricted? Could you figure that?

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:28]: Add a preferable slot, `:2` or `:3`?

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:36]: Drop these comments from the final ebuild.

**Comment by juippis** [comment]
You'll also need to properly sign-off all of your commits, 
https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin

and it'd be nice if you could use the syntax for our git commits, 
https://www.gentoo.org/glep/glep-0066.html#commit-messages but we can fix it for you.
Check https://gitweb.gentoo.org/repo/gentoo.git/log/ for examples.

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (misanthropos) [reply x11-misc/revelation/revelation-0.5.3.ebuild:4]: without gnome2.eclass gnome2_src_install wont work so I stick to EAPI 6 for the time being

File: x11-misc/revelation/Manifest
Comment (juippis) [inline x11-misc/revelation/Manifest:1]: Upstream seems to have released 0.5.4 meanwhile, let's update this PR to have 0.5.4 instead of 0.5.3.

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:4]:
```suggestion
EAPI=7
```

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:8]:
```suggestion
inherit python-single-r1 meson xdg
```

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:14]:
```suggestion
LICENSE="GPL-2+"
```

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:16]:
```suggestion
KEYWORDS="~amd64 ~x86"
```
With build system change, let's drop any old exotic arches.

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:21]:
```suggestion
# Upstream does not provide any test suite.
RESTRICT="test"
```

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:30]: Note that watch carefully the upstream dependency requirements, something is unlisted here. And you can use 
```
|| ( x1 x2 )
```
syntax to choose between *pycryptodome* and *pycryptodomex*.

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:37]:
```suggestion
```
(can be removed if you follow my instruction on enabling meson)

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:43]:
```suggestion
```
(can be removed if you follow my instruction on how to enable meson build)

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.3.ebuild:48]:
```suggestion
src_install() {
	meson_src_install
	python_fix_shebang "${ED}"
	python_optimize
}
```

**Review by juippis — COMMENTED** [review]
Now I've suggested many changes to bring the ebuild up-to-date and make it perfect. Please read through them carefully and make sure the end result works for you before committing changes here.

File: x11-misc/revelation/revelation-0.5.3.ebuild
Comment (juippis) [reply x11-misc/revelation/revelation-0.5.3.ebuild:30]: https://github.com/mikelolasagasti/revelation/blob/2a0b42276377fa7c44ac2e510f86ac33d36aa6b6/meson.build#L17 Here's the dep list.

**Comment by misanthropos** [comment]
there was actually no choice. with dev-python/pycryptodome the corresponding Cryptodome won`t be found.

**Comment by misanthropos** [comment]
I left 0.5.3 for now because 0.5.4 is not working.

**Review by misanthropos — COMMENTED** [review]
I have added 0.5.4 but left 0.5.3 because the former does not work because of a missing schema.

**Comment by misanthropos** [comment]
So this is glib related (I guess) / gschemas.compiled will be in /usr/share/glib-2.0/schemas but revelation expects it in /usr/share/revelation/schemas - I tried but can not figure out why that is.

File: x11-misc/revelation/revelation-0.5.4.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.4.ebuild:14]: Github is really bad at showing license files, you want to check the source files and their headers that are a bit more precise:
https://github.com/mikelolasagasti/revelation/blob/2a0b42276377fa7c44ac2e510f86ac33d36aa6b6/src/lib/util.py

File: x11-misc/revelation/metadata.xml
Comment (juippis) [reply x11-misc/revelation/metadata.xml:11]: This is still not done.

File: x11-misc/revelation/revelation-0.5.4.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.4.ebuild:44]:
```suggestion
}

pkg_postinst() {
	xdg_pkg_postinst
	gnome2_schemas_update
}

pkg_postrm() {
	xdg_pkg_postrm
	gnome2_schemas_update
}

```

File: x11-misc/revelation/revelation-0.5.4.ebuild
Comment (juippis) [inline x11-misc/revelation/revelation-0.5.4.ebuild:8]:
```suggestion
inherit gnome2-utils python-single-r1 meson xdg
```

File: x11-misc/revelation/files/revelation-0.5.4-undo-DIR_GSCHEMAS.patch
Comment (juippis) [inline x11-misc/revelation/files/revelation-0.5.4-undo-DIR_GSCHEMAS.patch:10]: Commenting is not needed since the patch file will show the line you've removed.

File: x11-misc/revelation/revelation-0.5.4.ebuild
Comment (misanthropos) [reply x11-misc/revelation/revelation-0.5.4.ebuild:40]: Thank you very much! That solved it without a patch.

**Comment by misanthropos** [comment]
I think its perfect now :)

**Review by juippis — APPROVED** [review]
Thanks for your patience and welcome aboard!

**Pull request closed without merging**
